pyaedt/emit.py

```diff
@@ -1,16 +1,242 @@
+        band_name : str
+           Name of the band.
+        tx_rx : tx_rx_mode object
+            Used for determining whether to get ``rx`` or ``tx`` radio names.
+        -------
+        :class:`list of float`
+
+        Examples
+        ----------
+        >>> bands = aedtapp.results.get_band_names('Bluetooth', 'Rx - Base Data Rate', Emit.tx_rx_mode.rx)
+        """
+        if self.get_result_loaded():
+            freq = self.emit_api.active_frequencies(radio_name, band_name, tx_rx_mode)
+        else:
+            freq = None
+            Emit.result_mode_error()
+        return freq
+
+
+class Revision:
+    """
+    Provides the ``Revision`` object.
+
+    Parameters
+    ----------
+    Emit_obj :
+         ``Emit`` object that this revision is associated with.
+    name : str, optional
+        Name of the revision to create. The default is ``None``, in which case a
+        default name is given.
+    Examples
+    --------
+    Create an instance of Revision.
+    >>> aedtapp = Emit()
+    >>> rev = Revision(aedtapp, "Revision 1")
+    >>> domain = Result.interaction_domain()
+    >>> rev.run(domain)
+    """
+
+    @pyaedt_function_handler()
+    def __init__(self, emit_obj, name=""):
+        subfolder = ""
+        for f in os.scandir(emit_obj.results.location):
+            if os.path.splitext(f.name)[1].lower() in ".aedtresults":
+                subfolder = os.path.join(f.path, "EmitDesign1")
+        default_behaviour = not os.path.exists(os.path.join(subfolder, "{}.emit".format(name)))
+        if default_behaviour:
+            print("The most recently generated revision will be used because the revision specified does not exist.")
+        if name == "" or default_behaviour:
+            file = max([f for f in os.scandir(subfolder)], key=lambda x: x.stat().st_mtime)
+            full = file.path
+            name = file.name
+        else:
+            full = subfolder + "/{}.emit".format(name)
+        self.name = name
+        self.path = full
+        self.emit_obj = emit_obj
+
+    @pyaedt_function_handler()
+    def run(self, domain):
+        """
+        Load the revision and then analyze along the given domain.
+
+        Parameters
+        ----------
+        domain :
+            ``InteractionDomain`` object for constraining the analysis parameters.
+        Returns
+        -------
+        :class:`Interaction`
+            Interaction object.
+
+        Examples
+        ----------
+        >>> domain = Result.interaction_domain()
+        >>> rev.run(domain)
+
+        """
+        self.emit_obj._load_revision(self.path)
+        eng = self.emit_obj._emit_api.get_engine()
+        interaction = eng.analyze(domain)
+        return interaction
 
 
 class Emit(FieldAnalysisEmit, object):
     """Provides the Emit application interface.
 
-    .. note::
-       This class creates only a skeleton for an empty design.
-       It has very limited functionalities, and no methods
-       are implemented yet.
```

Comment:
Google style guidelines say to write timeless documentation. In this case, that means removing the "yet" in line 12.

Reply:
I think this section was eliminated, but I'll keep that in mind